### PR TITLE
Update dry.md

### PR DESCRIPTION
**Pull request by johnblust**
Updates DRY code glossary page content

### CODE REVIEW

File: website/docs/terms/dry.md [hunk 1]
```diff
@@ -2,14 +2,89 @@
...
 :::
 
-Don’t Repeat Yourself - a coding ideal to work towards that can be applied at several levels: a variable, definition of business or transformation logic, joining two models, etc. 
+## Tools to help you write DRY code
+
+Let’s just say it: Writing DRY code is easier said than done. For classical software engineers, there’s a ton of resources out there to help them write DRY code. In the world of data transformation, there are also some tools and methodologies that can help folks in [the field of analytics engineering](https://www.getdbt.com/what-is-analytics-engineering/) write more DRY and modular code.
```
Comment (KiraFuruichi) [inline website/docs/terms/dry.md]:
```suggestion
Let’s just say it: Writing DRY code is easier said than done. For classical software engineers, there’s a ton of resources out there to help them write DRY code. In the world of data transformation, there are also some tools and methodologies that can help folks in [the field of analytics engineering](https://www.getdbt.com/what-is-analytics-engineering/) write more DRY and [modular code](https://www.getdbt.com/analytics-engineering/modular-data-modeling-technique/).
```

File: website/docs/terms/dry.md [hunk 2]
```diff
@@ -2,14 +2,89 @@
...
+
+
+###  Common Table Expressions (CTEs)
+
+CTEs are a great way to help you write more DRY code in your data analysis and dbt models. In a formal sense, a CTE is a temporary results set that can be used in a query. In a much more human and practical sense, we like to think of CTEs as separate, smaller queries within the larger query you’re building up. Essentially, you can use CTEs to break up complex queries into simpler blocks of code that are easier to debug and can connect and build off of each other.
```
Comment (KiraFuruichi) [inline website/docs/terms/dry.md]: Can we add a link/hover to CTE glossary page here?

File: website/docs/terms/dry.md [hunk 3]
```diff
@@ -2,14 +2,89 @@
...
+
+If you’re referencing a specific query, perhaps for aggregations that join back to an unaggregated view, CTEs can simply be referenced throughout a query with its CTE_EXPRESSION_NAME.
+
+
+### View materializations
+
+View [materializations](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/materializations) are also extremely useful for abstracting code that might otherwise be repeated often. A view is a defined passthrough SQL query that can be run against a database. Unlike a table, it doesn’t store data, but it defines the logic that you need to use to fetch the underlying data.
```
Comment (KiraFuruichi) [inline website/docs/terms/dry.md]: Can we link/hover to View glossary page here?

File: website/docs/terms/dry.md [hunk 4]
```diff
@@ -2,14 +2,89 @@
...
+
+If you’re referencing the same query, CTE, or block of code, throughout multiple data models, that’s probably a good sign that code should be its own view.
+
+For example, you might define a SQL view to count new users created in a day:
+
+```sql
+  select
+    created_date,
+    count(distinct(user_id)) as new_users
+  from {{ ref(‘users’) }}
```
Comment (KiraFuruichi) [inline website/docs/terms/dry.md]:
```suggestion
  from {{ ref('users') }}
```

**Review by KiraFuruichi — CHANGES_REQUESTED** [review]
Hell yessss looks great, thanks for getting this up! Left a few minor edits.

**Comment by johnblust** [comment]
Thanks for the review, Kira! Should be good to go now

**Review by KiraFuruichi — APPROVED** [review]

**Pull request merged**